kroxylicious-filter-test-support/src/main/java/io/kroxylicious/test/record/RecordTestUtils.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.test.record;
+
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+
+import org.apache.kafka.common.header.Header;
+import org.apache.kafka.common.record.CompressionType;
+import org.apache.kafka.common.record.MemoryRecords;
+import org.apache.kafka.common.record.MemoryRecordsBuilder;
+import org.apache.kafka.common.record.Record;
+import org.apache.kafka.common.record.RecordBatch;
+import org.apache.kafka.common.record.TimestampType;
+
+public class RecordTestUtils {
```

Comment:
Did you have some other cases in mind beyond factory methods? Maybe `RecordFactory` describes it better at the moment, or we could use `Records`.

Reply:
Not really. Right now the intention is that these methods are just for tests. Class names like `RecordFactory` make it less obvious that it's not really a general purpose thing. So I'd rather keep the ugly name.